internal/tasks/manifests.go

```diff
@@ -891,3 +893,178 @@ func (j *Janitor) setManifestAndParentsToPending(ctx context.Context, manifestDi
 
 	return err
 }
+
+var securityCheckSelectQuery = sqlext.SimplifyWhitespace(`
+	SELECT * FROM trivy_security_info
+	WHERE next_check_at <= $1
+	-- manifests without any check first, then sorted by schedule, then sorted by digest for deterministic behavior in unit test
+	ORDER BY next_check_at IS NULL DESC, next_check_at ASC, digest ASC
+	-- only one manifests at a time
+	LIMIT 1
+	-- prevent other job loops from working on the same asset concurrently
+	FOR UPDATE SKIP LOCKED
+`)
+
+func (j *Janitor) CheckTrivySecurityStatus(registerer prometheus.Registerer) jobloop.Job {
+	return (&jobloop.TxGuardedJob[*gorp.Transaction, keppel.TrivySecurityInfo]{
+		Metadata: jobloop.JobMetadata{
+			ReadableName: "check trivy security status",
+			CounterOpts: prometheus.CounterOpts{
+				Name: "keppel_trivy_security_status_checks",
+				Help: "Counter for Trivy security checks runs in manifests.",
+			},
+		},
+		BeginTx: j.db.Begin,
+		DiscoverRow: func(tx *gorp.Transaction, _ prometheus.Labels) (securityInfo keppel.TrivySecurityInfo, err error) {
+			err = tx.SelectOne(&securityInfo, securityCheckSelectQuery, j.timeNow())
+			return securityInfo, err
+		},
+		ProcessRow: j.processTrivySecurityInfo,
+	}).Setup(registerer)
+}
+
+func (j *Janitor) processTrivySecurityInfo(tx *gorp.Transaction, securityInfo keppel.TrivySecurityInfo, labels prometheus.Labels) error {
+	//load corresponding repo, account and manifest
+	repo, err := keppel.FindRepositoryByID(tx, securityInfo.RepositoryID)
+	if err != nil {
+		return fmt.Errorf("cannot find repo for manifest %s: %s", securityInfo.Digest, err.Error())
```

Comment:
When an underlying error is wrapped, %w should be used to preserve the inner error type.

```suggestion
		return fmt.Errorf("cannot find repo for manifest %s: %w", securityInfo.Digest, err)
```

Same for the other error returns in this method.